src/ddscxx/include/org/eclipse/cyclonedds/core/cdr/entity_properties.hpp

```diff
@@ -18,7 +18,14 @@
 #include <map>
 #include <atomic>
 #include <mutex>
+#if __cplusplus >= 201703L // c++17
+#define STD_IMPL std
```

Comment:
IIRC this file ends up in application code so we need to be carefully with macro names. `DDSCXX_STD_IMPL` seems a lot safer.

Alternatively (but here my limited knowledge of C++ puts me on rather shaky ground), perhaps we could do
```C++
#if __cplusplus >= 201703L
namespace std_impl = std;
#else
namespace std_impl = boost;
#endif
```
inside the `org::eclipse::cyclonedds::core::cdr` namespace? (So roughly where `decl_ref_type` is defined — and this makes me think that `decl_ref_type` itself probably slipped through the net ... but that's a different story.)